### PR TITLE
Cloning filter documenation

### PR DESCRIPTION
**Pull request by trevorhardy**
### Summary

Updated Configuration Options Reference to correctly show to configure a cloning filter.

### CODE REVIEW

File: docs/references/publications.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+## HELICS Application papers
+
+The following are a list of some of the published work using HELICS. If you have published work using HELICS you would like to see on this list, email support@helics.org with the citation and we'll review it.
+
+J. R. Theisen, A. Bose, M. Mukherjee, D. Burgess, K. Wilhelm and M. Diedesch, "Community-Based Transactive Coordination Mechanism for Enabling Grid-Edge Systems," 2024 IEEE Texas Power and Energy Conference (TPEC), College Station, TX, USA, 2024, pp. 1-6, doi: 10.1109/TPEC60005.2024.10472237, available at [https://ieeexplore.ieee.org/document/10472237](https://ieeexplore.ieee.org/document/10472237).
+
+J. R. Theisen, A. Bose, M. Mukherjee, D. Burgess, K. Wilhelm and M. Diedesch, "Community-Based Transactive Coordination Mechanism for Enabling Grid-Edge Systems," 2024 IEEE Texas Power and Energy Conference (TPEC), College Station, TX, USA, 2024, pp. 1-6, doi: 10.1109/TPEC60005.2024.10472237 available at [https://ieeexplore.ieee.org/document/10472237](https://ieeexplore.ieee.org/document/10472237).
```
Comment (nightlark) [inline docs/references/publications.md:25]: This looks like a duplicate of the previous citation.

Comment (nightlark) [inline docs/references/publications.md:25]:
```suggestion
```

This citation looks like a duplicate of the previous citation.

File: docs/user-guide/advanced_topics/connector_fed.md [hunk 2]
```diff
@@ -6,24 +6,23 @@ To support this HELICS comes with a connector application called "helics_connect
...
+The order in which "helics*connector" creates the interfaces in the federate is not deterministic. This is especially important when using the APIs that get the interface by index (\_e.g* `helicsFederateGetInputByIndex)`, `helicsFederateGetPublicationByIndex`, `helicsFederateGetEndpointByIndex`, `helicsFederateGetEndpointByIndex`). There is no guarantee that the order the interfaces are added to the federate will be the same from computer to computer. To continue using this API
 
 ## Examples
 
 Three examples have been created to demonstrate the use of the "helics_connector" federate: one uses a direct match-file, one uses a regex match-file and the the last uses a direct match-file while using the query-and-command method of interface creation. Links to the examples are provided below; all produce identical results.
 
-An example utilizing both kinds of iteration will be developed shortly. Iteration example is available in [here](../examples/advanced_examples/advanced_iteration.md)
+An example utilizing both kinds of iteration is available in [here](../examples/advanced_examples/advanced_iteration.md)
```
Comment (nightlark) [inline docs/user-guide/advanced_topics/connector_fed.md:28]:
```suggestion
An example utilizing both kinds of iteration is available [here](../examples/advanced_examples/advanced_iteration.md)
```

File: docs/user-guide/advanced_topics/connector_fed.md [hunk 3]
```diff
@@ -6,24 +6,23 @@ To support this HELICS comes with a connector application called "helics_connect
...
 
 ## "helics_connector" Interface creation
 
-The order in which "helics*connector" specified the needed interfaces in the federate is not deterministic. This is especially important when using the APIs that get the interface by index (\_e.g* `helicsFederateGetInputByIndex)`, `helicsFederateGetPublicationByIndex`, `helicsFederateGetEndpointByIndex`, `helicsFederateGetEndpointByIndex`). There is no guarantee that the order the interfaces are added to the federate will be the same from computer to computer. To continue using this API the name should be queried then potentially reordered or use the `ByName` version of the calls.
+The order in which "helics*connector" creates the interfaces in the federate is not deterministic. This is especially important when using the APIs that get the interface by index (\_e.g* `helicsFederateGetInputByIndex)`, `helicsFederateGetPublicationByIndex`, `helicsFederateGetEndpointByIndex`, `helicsFederateGetEndpointByIndex`). There is no guarantee that the order the interfaces are added to the federate will be the same from computer to computer. To continue using this API
```
Comment (nightlark) [inline docs/user-guide/advanced_topics/connector_fed.md:22]:
```suggestion
The order in which "helics*connector" creates the interfaces in the federate is not deterministic. This is especially important when using the APIs that get the interface by index (\_e.g* `helicsFederateGetInputByIndex)`, `helicsFederateGetPublicationByIndex`, `helicsFederateGetEndpointByIndex`, `helicsFederateGetEndpointByIndex`). There is no guarantee that the order the interfaces are added to the federate will be the same from computer to computer. To continue using this API.
```

File: docs/user-guide/examples/advanced_examples/advanced_async_time_request.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+There is also a similar set of APIs when working with the iterative time requests: `helicsFederateRequestTimeIterativeAsync()` and `helicsFederateRequestTimeIterativeComplete()`.
+
+## Example Explanation
+
+The advanced default example has been very slightly modified to demonstrate the use of `helicsFederateRequestTimeAsync()` and `helicsFederateRequestTimeComplete()`. In this case, Battery.py has been edited to always make the async time request and then immediately call a frivalous function that, once a simulated day, delays the time request by one wall-clock second and sends a message indicating to the logs. (Obviously, in a real-world application the work down in this function would not be frivalous.) Running this example and looking at the logs shows this message, indicating the function is being run after the async time request but before completing the time requeest.
```
Comment (nightlark) [inline docs/user-guide/examples/advanced_examples/advanced_async_time_request.md:15]:
```suggestion
The advanced default example has been very slightly modified to demonstrate the use of `helicsFederateRequestTimeAsync()` and `helicsFederateRequestTimeComplete()`. In this case, Battery.py has been edited to always make the async time request and then immediately call a frivolous function that, once a simulated day, delays the time request by one wall-clock second and sends a message to the logs. (Obviously, in a real-world application the work done in this function would not be frivolous.) Running this example and looking at the logs shows this message, indicating the function is being run after the async time request but before completing the time request.
```

File: docs/user-guide/examples/advanced_examples/advanced_async_time_request.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+There are cases, though, where a federate may want to make a time request and then continue working on another computational task instead of waiting in an idle state for the time request to be granted. HELICS provides an alternative time request API to accomplish this: `helicsFederateRequestTimeAsync()`.
+
+## Using the Asynchronous Time Request
+
+Making the asynchronous time request is simple: to make the non-blocking time request use the `helicsFederateRequestTimeAsync()` API. This API does not return a value and does not block the federate operation. The code following this API will begin immediately executing while the rest of the federation advance through simulated time. When this federate is done with it's extra work, is uses the `helicsFederateRequestTimeComplete()` API call. This is a blocking call (just like `hliecsFederateRequestTime()` was) though it is possible the state of the federation is such that HELICS is ready to immediately grant a time.
```
Comment (nightlark) [inline docs/user-guide/examples/advanced_examples/advanced_async_time_request.md:9]:
```suggestion
Making the asynchronous time request is simple: to make the non-blocking time request use the `helicsFederateRequestTimeAsync()` API. This API does not return a value and does not block the federate operation. The code following this API will begin immediately executing while the rest of the federation advances through simulated time. When this federate is done with its extra work, it uses the `helicsFederateRequestTimeComplete()` API call. This is a blocking call (just like `hliecsFederateRequestTime()` was), though it is possible the state of the federation is such that HELICS is ready to immediately grant a time.
```

File: docs/user-guide/examples/advanced_examples/advanced_async_time_request.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
+# Asynchronous (non-blocking) Time Request
+
+Generally, when federates make a time request they do so by calling helicsFederateRequestTime(). This is a blocking call; the HELICS library will wait until it is has determined the time it will grant the federate and then will do so, returning the granted time. In many cases, this blocking call is acceptable as a time request generally means a federate has nothing to do until it receives new inputs from the rest of the federation as it has completed updating its internal model based on the last received inputs from the federation.
```
Comment (nightlark) [inline docs/user-guide/examples/advanced_examples/advanced_async_time_request.md:3]:
```suggestion
Generally, when federates make a time request they do so by calling `helicsFederateRequestTime()`. This is a blocking call; the HELICS library will wait until it has reached the time it determines should be granted to the federate and then will do so, returning the granted time. In many cases, this blocking call is acceptable as a time request generally means a federate has nothing to do until it receives new inputs from the rest of the federation as it has completed updating its internal model based on the last received inputs from the federation.
```

File: docs/references/configuration_options_reference.md [hunk 7]
```diff
@@ -1255,14 +1244,11 @@ The `info` field is entirely ignored by HELICS and is used as a mechanism to pas
...
 
 Filters are registered with the core or through the application API.
-There are also Filter object that hide some of the API calls in a slightly nicer interface. Generally a filter will define a target endpoint as either a source filter or destination filter. Source filters can be chained, as in there can be more than one of them. At present there can only be a single non-cloning destination filter attached to an endpoint.
+There are also Filter object that hide some of the API calls in a slightly nicer interface. Generally a filter will be associated with a specific endpoint as either a source filter or destination filter. Source filters can be chained, as in there can be more than one of them. At present there can only be a single non-cloning destination filter attached to an endpoint.
 
-Non-cloning filters can modify the message in some ways, cloning filters just copy the message and may send it to multiple destinations.
+Non-cloning filters can modify the message in some ways, cloning filters just copy the message and may send it to multiple destinations. Cloning not considered the "operation" of the filter and can be specified in parallel to another filter operation.
```
Comment (nightlark) [inline docs/references/configuration_options_reference.md:1249]:
```suggestion
Non-cloning filters can modify the message in some ways, cloning filters just copy the message and may send it to multiple destinations. Cloning is not considered the "operation" of the filter and can be specified in parallel to another filter operation.
```

File: docs/conf.py [hunk 8]
```diff
@@ -40,6 +40,15 @@
 
 current_directory = os.path.dirname(os.path.realpath(__file__))
 
+# Updating definitions so that add-ons load correctly. These changes will
+# be required by Oct 7, 2024
+# See https://about.readthedocs.com/blog/2024/07/addons-by-default/
+# Define the canonical URL if you are using a custom domain on Read the Docs
+html_baseurl = os.environ.get("READTHEDOCS_CANONICAL_URL", "")
```
[the remark this answers is not in the document]
Comment (trevorhardy) [reply docs/conf.py:47]: I don't know. Let's make the change and see.

**Comment by trevorhardy** [comment]
Is this branch getting the documentation built? When I try to build locally I'm getting some errors and I don't want to make sure the RTD version is showing the same thing.

**Comment by nightlark** [comment]
All PRs to main or develop get documentation built.

**Comment by trevorhardy** [comment]
> All PRs to main or develop get documentation built.

Can I access those builds from the RTD interface? I'm not seeing them listed as alternative versions there.

**Comment by nightlark** [comment]
Look at the build statuses on the PR for a link.

File: docs/user-guide/advanced_topics/advanced_topics_index.md [hunk 9]
```diff
@@ -70,12 +66,6 @@ The Advanced Topics will cover:
 - [**Simultaneous co-simulations**](./simultaneous_cosimulations.md) - Options for running multiple independent co-simulations on a single system.
 - [**Targeted Endpoints**](./targeted_endpoints.md) - details on the new targeted endpoints in HELICS 3.
 - [**Timeouts**](./timeouts.md) - HELICS includes a number of timeouts to prevent failed operations from continuing indefinitely, the various timeout options are discussed in this document.
-- [**Command Interface**](./commandInterface.md) - HELICS v3 introduced the command interface as a method of asynchronously communicating between federates.
-- [**Aliases**](./aliases.md) - HELICS 3.3 introduced the notion of aliases. Aliases allow a mapping of an interface key to a different string.
```
Comment (phlptp) [inline docs/user-guide/advanced_topics/advanced_topics_index.md:74]: Did we move these links somewhere?

**Review by phlptp — APPROVED** [review]

**Pull request merged**